source/initial_temperature/random_gaussian_perturbation.cc

```diff
@@ -0,0 +1,208 @@
+
+          for (unsigned int d=0; d<dim; ++d)
+            {
+              min_coordinates[d] = -spherical_geometry_model.outer_radius();
+              max_coordinates[d] = spherical_geometry_model.outer_radius();
+            }
+        }
+      else if (Plugins::plugin_type_matches<const GeometryModel::Sphere<dim>> (this->get_geometry_model()))
+        {
+          const GeometryModel::Sphere<dim> &sphere_geometry_model =
+            Plugins::get_plugin_as_type<const GeometryModel::Sphere<dim>> (this->get_geometry_model());
+
+          for (unsigned int d=0; d<dim; ++d)
+            {
+              min_coordinates[d] = -sphere_geometry_model.radius();
+              max_coordinates[d] = sphere_geometry_model.radius();
+            }
+        }
+      else if (Plugins::plugin_type_matches<const GeometryModel::Chunk<dim>> (this->get_geometry_model()))
+        {
+          const GeometryModel::Chunk<dim> &chunk_geometry_model =
+            Plugins::get_plugin_as_type<const GeometryModel::Chunk<dim>> (this->get_geometry_model());
+
+          for (unsigned int d=0; d<dim; ++d)
+            {
+              min_coordinates[d] = -chunk_geometry_model.outer_radius();
+              max_coordinates[d] = chunk_geometry_model.outer_radius();
+            }
+        }
+      else if (Plugins::plugin_type_matches<const GeometryModel::Box<dim>> (this->get_geometry_model()))
+        {
+          const GeometryModel::Box<dim> &box_geometry_model =
+            Plugins::get_plugin_as_type<const GeometryModel::Box<dim>> (this->get_geometry_model());
+
+          min_coordinates = box_geometry_model.get_origin();
+          max_coordinates = min_coordinates + box_geometry_model.get_extents();
+        }
+      else if (Plugins::plugin_type_matches<const GeometryModel::TwoMergedBoxes<dim>> (this->get_geometry_model()))
+        {
+          const GeometryModel::TwoMergedBoxes<dim> &two_merged_boxes_geometry_model =
+            Plugins::get_plugin_as_type<const GeometryModel::TwoMergedBoxes<dim>> (this->get_geometry_model());
+
+          min_coordinates = two_merged_boxes_geometry_model.get_origin();
+          max_coordinates = min_coordinates + two_merged_boxes_geometry_model.get_extents();
+        }
+      else
+        AssertThrow (false,
+                     ExcMessage ("Not a valid geometry model for the initial conditions model "
+                                 "random Gaussian perturbation."));
+
+      // Initialize random locations of perturbations and check they are in the model domain.
+      // Use a fixed number as seed for random generator.
+      // This is important if we run the code on more than 1 processor.
+      std::srand(1);
+
+      for (unsigned int n=0; n<n_perturbations; ++n)
+        {
+          do
+            {
+              for (unsigned int d=0; d<dim; ++d)
+                perturbation_centers[n][d] = min_coordinates[d] + (double)(std::rand() % int(max_coordinates[d] - min_coordinates[d]));
```

Comment:
Please change to this (easier and using our own generator):
```suggestion
                perturbation_centers[n][d] = min_coordinates[d] + random_location(generator) * (max_coordinates[d] - min_coordinates[d]);
```